datalad/distributed/ora_remote.py

```diff
@@ -1023,9 +1032,23 @@ def initremote(self):
         if not isinstance(self.io, HTTPRemoteIO):
             self.get_store()
 
-        # else:
-        # TODO: consistency with SSH and FILE behavior? In those cases we make
-        #       sure the store exists from within initremote
+        else:
+            # TODO: Consistency with SSH and FILE behavior? In those cases we
+            #       make sure the store exists from within initremote. In case
+            #       of HTTP, we can't do that much, since it's up to the server
+            #       how to satisfy our annex key requests. Assuming an
+            #       underlying structure on FS and validate it isn't really
+            #       possible. However, may be we shouldn't do that much for SSH
+            #       and Local IO either.
+            #
+            #       In any case, do a minimal check for whether we can connect
+            #       to the store, since initremote/enableremote have to fail
+            #       otherwise. Note, that we shouldn't expect the webserver to
+            #       serve things like ria-layout-version file. This is mainly to
+            #       verify that we can in principle connect via that URL.
+            if not self.io.exists('HEAD'):
```

Comment:
* It looks like `.exists` will fail if this isn't a pathlib object

 * Is it safe to assume the bare repo `HEAD` exists in a store?  It won't with, e.g., stores created with `create-sibling-ria --storage-sibling=only`, right?

Reply:
> Is it safe to assume the bare repo HEAD exists in a store? It won't with, e.g., stores created with create-sibling-ria --storage-sibling=only, right?

Good point, forgot we allow for the absence of a git repo. All that is actually required for the remote to work is correct response to requests (HEAD|GET) to annex keys. But for initremote/enableremote there's no key we could generally assume to be there. So, I wonder what to test for. We could execute a checkpresent for an arbitrary key. Issue is: What does a 404 really tell us? We can tell host was resolved and there's a webserver responding. That's at least something. On the other hand that doesn't verify we can get anything related to that dataset.
May be we need to just define something else, a valid RIA store needs to serve over HTTP. `ria-layout-version` (at least at dataset level, not necessarily top-level) would be an obvious choice that would also increase flexibility for URL building (mapping to a different layout could then be done client-side just like with SSH/local). What do you think, @mih?

Reply:
I hope I was not blocking this PR for 5 months, because I have nothing substantial to say. #5459 is rather terse and does not have any information on the context of the error. Taking what is provided it is unclear to me what `initremote` could do to prevent a future `enableremote` from failing. A server could vanish any time, or become misconfigured. To me this sounds like a UX of `enableremote` rather than a problem of `initremote`. Of course it would be good to check sanity of URLs given to `initremote`.

Looking at the code at the code I see that such checks are made in ~`get_store()`, but are explicitly disabled for HTTP access in `initremote()`. I'd say that it needs no change in behavior or logic, just a removal of the shortcuts made for HTTP access.

Reply:
@mih : There's no difference between `initremote` and `enableremote` from the point of view of a special remote. `INITREMOTE` is what it gets from annex it both cases.

> Looking at the code at the code I see that such checks are made in ~get_store(), but are explicitly disabled for HTTP access in initremote(). I'd say that it needs no change in behavior or logic, just a removal of the shortcuts made for HTTP access.

Yes and no. Needs an equivalent. But in opposition to local/SSH access we don't really need the same things to be checked, since the special remote doesn't actually operate on the store's FS tree.

I will change this to request the `ria-layout-version` files both from the store- and dataset level. This will be a more general change, as ORA will then use that file to determine the key URL, just like in the case of local FS and SSH. So, abandon the idea that we approach it differently via HTTP in that we only ever send uniform requests independent of any `version` f the store and leave it to the server to map the requests differently, if needed. Instead we get that version and build the URL from it exactly like we'd do it for non-HTTP access.
This not only gives us something to verify but also enables another change in the future: A version file, that doesn't necessarily give a number/label referring to a predefined-in-datalad layout, but alternatively specifies the layout directly, so that a store's maintainer has all the flexibility to set it up w/o a need for the special remote to know the layout. I wanted to go there anyway, so I think this is going to solve more than one problem.

Reply:
I am resolving this old conversation as it seems no longer relevant.